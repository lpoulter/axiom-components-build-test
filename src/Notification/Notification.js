import atIds from '@brandwatch/axiom-automation-testing/ids';
import PropTypes from 'prop-types';
import React, { Component } from 'react';
import classnames from 'classnames';
import omit from 'lodash.omit';
import { ALERT_ICON_NAME_MAP } from '../AlertIcon/AlertIcon';
import Base from '../Base/Base';
import Icon from '../Icon/Icon';
import Link from '../Typography/Link';
import './Notification.css';

export default class Notification extends Component {
  static propTypes = {
    children: PropTypes.node.isRequired,
    onAppear: PropTypes.func,
    onRemoveClick: PropTypes.func,
    type: PropTypes.oneOf([
      'error',
      'info',
      'success',
      'warning',
    ]).isRequired,
  };

  componentDidMount() {
    const { onAppear } = this.props;

    if (onAppear) {
      onAppear();
    }
  }

  render() {
    const { children, type, onRemoveClick, ...rest } = this.props;
    const classes = classnames('ax-notification', `ax-notification--${type}`);

    return (
      <Base { ...omit(rest, ['onAppear']) } className={ classes } textColor="subtle">
        <div className="ax-notification__icon">
          <Icon name={ ALERT_ICON_NAME_MAP[type] } size="2rem" />
        </div>

        <div className="ax-notification__messsage">
          { children }
        </div>

        { onRemoveClick && (
          <div className="ax-notification__remove">
            <Link
                data-ax-at={ atIds.Notification.close }
                onClick={ () => onRemoveClick() }
                style="subtle">
              <Icon name="cross" />
            </Link>
          </div>
        ) }
      </Base>
    );
  }
}
