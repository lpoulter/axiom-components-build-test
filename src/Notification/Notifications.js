import PropTypes from 'prop-types';
import React, { Component, Children } from 'react';
import Base from '../Base/Base';
import Portal from '../Portal/Portal';

export default class Notifications extends Component {
  static propTypes = {
    children: PropTypes.node,
  };

  render() {
    const { children, ...rest } = this.props;
    const isRendered = Boolean(Array.isArray(children)
      ? children.length
      : children);

    if (!isRendered) {
      return null;
    }

    return (
      <Portal>
        <Base { ...rest } className="ax-notifications">
          { Children.map(children, (child) =>
            <div className="ax-notifications__notification">
              { child }
            </div>
          ) }
        </Base>
      </Portal>
    );
  }
}
