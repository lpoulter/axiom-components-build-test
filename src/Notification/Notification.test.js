import React from 'react';
import renderer from 'react-test-renderer';
import Notification from './Notification';

function getComponent(props = {}) {
  return renderer.create(
    <Notification { ...props }>
      Lorem ipsum
    </Notification>
  );
}

describe('Notification', () => {
  it('renders with defaultProps', () => {
    const component = getComponent({ type: 'info' });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with onRemoveClick', () => {
    const component = getComponent({ type: 'info', onRemoveClick: () => {} });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('renders with type', () => {
    ['error', 'info', 'success', 'warning'].forEach((type) => {
      it(type, () => {
        const component = getComponent({ type });
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });
});
