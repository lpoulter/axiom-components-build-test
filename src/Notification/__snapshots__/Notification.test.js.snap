// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification renders with defaultProps 1`] = `
<div
  className="ax-notification ax-notification--info ax-text--color-subtle"
>
  <div
    className="ax-notification__icon"
  >
    <svg
      className="ax-icon ax-icon--information"
      dangerouslySetInnerHTML={
        Object {
          "__html": "",
        }
      }
      style={
        Object {
          "height": "2rem",
          "width": "2rem",
        }
      }
      viewBox="0 0 16 16"
    />
  </div>
  <div
    className="ax-notification__messsage"
  >
    Lorem ipsum
  </div>
</div>
`;

exports[`Notification renders with onRemoveClick 1`] = `
<div
  className="ax-notification ax-notification--info ax-text--color-subtle"
>
  <div
    className="ax-notification__icon"
  >
    <svg
      className="ax-icon ax-icon--information"
      dangerouslySetInnerHTML={
        Object {
          "__html": "",
        }
      }
      style={
        Object {
          "height": "2rem",
          "width": "2rem",
        }
      }
      viewBox="0 0 16 16"
    />
  </div>
  <div
    className="ax-notification__messsage"
  >
    Lorem ipsum
  </div>
  <div
    className="ax-notification__remove"
  >
    <a
      className="ax-link ax-link--style-subtle"
      data-ax-at="166ebc0c761f"
      onClick={[Function]}
    >
      <svg
        className="ax-icon ax-icon--cross"
        dangerouslySetInnerHTML={
          Object {
            "__html": "",
          }
        }
        style={
          Object {
            "height": "1rem",
            "width": "1rem",
          }
        }
        viewBox="0 0 16 16"
      />
    </a>
  </div>
</div>
`;

exports[`Notification renders with type error 1`] = `
<div
  className="ax-notification ax-notification--error ax-text--color-subtle"
>
  <div
    className="ax-notification__icon"
  >
    <svg
      className="ax-icon ax-icon--warning"
      dangerouslySetInnerHTML={
        Object {
          "__html": "",
        }
      }
      style={
        Object {
          "height": "2rem",
          "width": "2rem",
        }
      }
      viewBox="0 0 16 16"
    />
  </div>
  <div
    className="ax-notification__messsage"
  >
    Lorem ipsum
  </div>
</div>
`;

exports[`Notification renders with type info 1`] = `
<div
  className="ax-notification ax-notification--info ax-text--color-subtle"
>
  <div
    className="ax-notification__icon"
  >
    <svg
      className="ax-icon ax-icon--information"
      dangerouslySetInnerHTML={
        Object {
          "__html": "",
        }
      }
      style={
        Object {
          "height": "2rem",
          "width": "2rem",
        }
      }
      viewBox="0 0 16 16"
    />
  </div>
  <div
    className="ax-notification__messsage"
  >
    Lorem ipsum
  </div>
</div>
`;

exports[`Notification renders with type success 1`] = `
<div
  className="ax-notification ax-notification--success ax-text--color-subtle"
>
  <div
    className="ax-notification__icon"
  >
    <svg
      className="ax-icon ax-icon--tick"
      dangerouslySetInnerHTML={
        Object {
          "__html": "",
        }
      }
      style={
        Object {
          "height": "2rem",
          "width": "2rem",
        }
      }
      viewBox="0 0 16 16"
    />
  </div>
  <div
    className="ax-notification__messsage"
  >
    Lorem ipsum
  </div>
</div>
`;

exports[`Notification renders with type warning 1`] = `
<div
  className="ax-notification ax-notification--warning ax-text--color-subtle"
>
  <div
    className="ax-notification__icon"
  >
    <svg
      className="ax-icon ax-icon--warning"
      dangerouslySetInnerHTML={
        Object {
          "__html": "",
        }
      }
      style={
        Object {
          "height": "2rem",
          "width": "2rem",
        }
      }
      viewBox="0 0 16 16"
    />
  </div>
  <div
    className="ax-notification__messsage"
  >
    Lorem ipsum
  </div>
</div>
`;
